test/index.js

```diff
@@ -460,4 +470,81 @@ describe('KeyringController', function () {
+      await keyringController.submitPassword(password);
+
+      expect(keyringController.password).toBe(password);
+      expect(keyringController.encryptionSalt).toBe('SALT');
+    });
+
+    it('unlocks the keyrings with valid information', async function () {
+      keyringController.cacheEncryptionKey = true;
+      const returnValue = await keyringController.encryptor.decryptWithKey();
+      const stub = sinon.stub(keyringController.encryptor, 'decryptWithKey');
+      stub.resolves(Promise.resolve(returnValue));
+
+      keyringController.store.updateState({ vault: MOCK_ENCRYPTION_DATA });
+      await keyringController.submitEncryptionKey(MOCK_ENCRYPTION_KEY);
+
+      expect(keyringController.encryptor.decryptWithKey.calledOnce).toBe(true);
+      expect(keyringController.keyrings).toHaveLength(1);
+    });
+
+    it('should not load keyrings when incorrect encryptionKey', async function () {
+      keyringController.cacheEncryptionKey = true;
+      keyringController.setLocked(true);
+
+      await expect(keyringController.submitEncryptionKey(`{}`)).rejects.toThrow(
+        `Failed to execute 'importKey' on 'SubtleCrypto': The provided value is not of type '(ArrayBuffer or ArrayBufferView or JsonWebKey)'.`,
+      );
+      expect(keyringController.password).toBeUndefined();
+      expect(keyringController.keyrings).toHaveLength(0);
+    });
+
+    it('persists keyrings when actions are performed', async function () {
+      keyringController.cacheEncryptionKey = true;
+      keyringController.store.updateState({ vault: MOCK_ENCRYPTION_DATA });
+      await keyringController.submitEncryptionKey(MOCK_ENCRYPTION_KEY);
+
+      const stub = sinon.stub(keyringController, 'persistAllKeyrings');
+      stub.resolves(Promise.resolve(true));
+
+      const [firstKeyring] = keyringController.keyrings;
+
+      await keyringController.addNewAccount(firstKeyring);
+      expect(stub.callCount).toBe(1);
+
+      await keyringController.addNewAccount(firstKeyring);
+      expect(stub.callCount).toBe(2);
+
+      const account = {
+        privateKey:
+          'c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3',
+        publicKey: '0x627306090abab3a6e1400e9345bc60c78a8bef57',
+      };
+
+      // Add a new keyring with one account
+      await keyringController.addNewKeyring('Simple Key Pair', [
+        account.privateKey,
+      ]);
+      expect(stub.callCount).toBe(3);
+
+      // remove that account that we just added
+      await keyringController.removeAccount(account.publicKey);
+      expect(stub.callCount).toBe(4);
+    });
+
+    it('cleans up login artifacts upon lock', async function () {
+      await keyringController.submitPassword(password);
+
+      await keyringController.setLocked();
+
+      expect(keyringController.encryptionSalt).toBeUndefined();
+      expect(keyringController.password).toBeUndefined();
+      expect(keyringController.memStore.getState().encryptionKey).toBeNull();
```

Comment:
The `cacheEncryptionKey` option isn't being set here, so this assertion isn't doing what it appears to be doing. It'll be `null` here not because it was cleared, but because it was never set.

Reply:
Good shout!  Updated!